### PR TITLE
chore: Migrate region module to use DML

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/modules/region/src/models/country.ts [hunk 1]
```diff
@@ -1,44 +1,19 @@
...
+    iso_3: model.text(),
+    num_code: model.text(),
+    name: model.text(),
+    display_name: model.text(),
+    region: model.belongsTo(() => Region, { mappedBy: "countries" }).nullable(),
+    metadata: model.json().nullable(),
+  }
+)
+
+// .indexes([{ fields: ["region_id", "iso_2"], unique: true, name: "IDX_region_country_region_id_iso_2_unique" ])
```
Comment (olivermrbl) [inline packages/modules/region/src/models/country.ts]: PR blocked by missing composite indexes API

File: packages/modules/region/src/models/country.ts [hunk 2]
```diff
@@ -1,44 +1,28 @@
...
-  region?: Region | null
-}
+const Country = model
+  .define(
+    { name: "Country", tableName: "region_country" },
+    {
+      iso_2: model.text().primaryKey(),
+      iso_3: model.text(),
+      num_code: model.text(),
+      name: model.text(),
```
Comment (sradevski) [inline packages/modules/region/src/models/country.ts]: Should the country code and name be searchable as well?

**Review by sradevski — APPROVED** [review]
Nice 👍

**Review by riqwan — APPROVED** [review]
nice!

File: packages/modules/region/src/models/country.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/modules/region/src/models/country.ts]: Yeah, I guess it doesn't hurt. Will add that in

**Pull request merged**